engine/chain/internal/bhs/get_merkleroots_from_bhs.go

```diff
@@ -0,0 +1,64 @@
+package bhs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"syscall"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+)
+
+// GetMerkleRootsFromBHS returns Merkle Roots from Block Header Service
+func (s *Service) GetMerkleRootsFromBHS(ctx context.Context, query url.Values) (*models.MerkleRootsBHSResponse, error) {
+	bhsURL, err := s.createBHSURL("/chain/merkleroot")
+	if err != nil {
+		return nil, err
+	}
+
+	req := s.httpClient.R().
+		SetContext(ctx).
+		SetHeader("Content-Type", "application/json").
+		EnableTrace()
+
+	if s.bhsCfg.AuthToken != "" {
+		req.SetAuthToken(s.bhsCfg.AuthToken)
+	} else {
+		s.logger.Warn().Msg("warning creating Block Headers Service url - auth token is not set. Some requests might not work")
+	}
+
+	var response models.MerkleRootsBHSResponse
+	res, err := req.
+		SetResult(&response).
+		SetQueryParamsFromValues(query).
+		Get(bhsURL.String())
```

Comment:
In the `chain` package I've used the convention in which the `Get/Post` method call is not mexed up with the `req` configuration (e.g. SetResult`).

After adjustments it would look like this:
```golang
        var response models.MerkleRootsBHSResponse
	req := s.httpClient.R().
		SetContext(ctx).
		SetHeader("Content-Type", "application/json").
		EnableTrace().
		SetResult(&response).
		SetQueryParamsFromValues(query)

	if s.bhsCfg.AuthToken != "" {
		req.SetAuthToken(s.bhsCfg.AuthToken)
	} else {
		s.logger.Warn().Msg("warning creating Block Headers Service url - auth token is not set. Some requests might not work")
	}

	
	res, err := req.Get(bhsURL.String())
```

Reply:
done